pkg/client/fake/client.go

```diff
@@ -123,6 +123,11 @@ func (c *fakeClient) Update(ctx context.Context, obj runtime.Object) error {
 	return c.tracker.Update(gvr, obj, accessor.GetNamespace())
 }
 
+func (c *fakeClient) UpdateStatus(ctx context.Context, obj runtime.Object) error {
+	// TODO(droot): implement for *real*
```

Comment:
we should prob implement this before merging.

Reply:
I am calling `c.Update` now which will update the Status, but will also update the Spec part, which is wrong. 

Still looking for a way to do it with underneath tracker.